### DISCUSSION TITLE
Serverside rendering?

### DISCUSSION THREAD

**Discussion by jmfk** [google/mesop#1115] [Ideas]
Don't even know it is possible? Or even partly possible? 
But rendering a HTML version of the page that could be sent (with or without the Angular code) would make this more useful for webpages.

**Comment by richard-to** [comment]
Angular does support SSR. However, I have never used Angular SSR before, so I'm not sure what would be needed for it to work. I think it's possible though. 

I think it probably wouldn't be possible to generate a server side version of the HTML without the Angular code and Mesop server.

**Reply by wwwillchen** [reply]
+1 to what @richard-to said - you really need the Angular app + Mesop server to support any kind of interactivity.

I expect this would be a fairly non-trivial effort given that it would introduce a node.js server dependency. I haven't yet tried this personally, but I think other Python UI frameworks like https://fastht.ml/ may be a better fit for those kinds of use cases, because IIUC, you construct an HTML directly on the server-side .

**Reply by jmfk** [reply]
Thanks Will. I'll check that out.
